src/components/LeaderBoard/TeamLeaderboardChart.tsx

```diff
@@ -153,42 +179,140 @@ const TeamLeaderBoardChart: React.FC<{ data: Array<Data> }> = ({ data }) => {
     }
   };
 
+  //ajusts the size of the element containing the chart in order to correctly size the chart.
+  const boxAdjust = (labelLength: number) => {
+    const box = document.querySelector('.team-box');
+    if (box != null) {
+      const newHeight = labelLength * chartHeightMultiplier;
+      box.setAttribute('style', 'height: ' + newHeight.toString() + 'px');
+    }
+  };
+
   //gives leaderboard placement numbers a suffix
   const ordinalNumbers = (n: number) => {
     return n > 0
       ? ['th', 'st', 'nd', 'rd'][(n > 3 && n < 21) || n % 10 > 3 ? 0 : n % 10]
       : '';
   };
-
-  //ajusts the size of the element containing the chart in order to correctly size the chart.
-  const boxAdjust = (labelLength: number) => {
-    const box = document.querySelector('.box-team-leaderboard');
-    if (box != null) {
-      const newHeight = labelLength * 60;
-      box.setAttribute('style', 'height: ' + newHeight.toString() + 'px');
+  //gets the index to calculate the scoll distance needed to bring the user into view
+  const scrollToUser = () => {
+    const content = contentRef.current;
+    let y = 0;
+    data.every((member: any) => {
+      if (!member.highlight) {
+        y += 1;
+        return true;
+      } else {
+        return false;
+      }
+    });
+    if (content) {
+      content.scrollTop = (y+2) * chartHeightMultiplier;
     }
   };
+  //gets the data from the db for users or teams, sorts them based on highest to lowest steps, and sets the data
+  async function getData(dataType: string) {
+    setLoading(true);
+    const indData: Array<Data> = [];
+    if (dataType == 'teamMembers') {
```

Comment:
If the dataType is teamMembers, we should use the passed-in parameter. We don't want to call getDoc here we need to minimize the use of getDoc function.

Reply:
What we have as a passed-in parameter here is a real-time update data from teamHome.tsx

Reply:
I have made some changes to the LeaderboardChart.tsx if you want to take another look, I still need to implement the same changes into the TeamLeaderboardChart.tsx